test/test_crud_v2.py

```diff
@@ -0,0 +1,331 @@
+
+            for attr, expected in expectation[event_type].items():
+                actual = getattr(event, attr)
+                if isinstance(expected, dict):
+                    for key, val in expected.items():
+                        if val is None:
+                            if key in actual:
+                                self.fail("Unexpected key [%s] in %r" % (
+                                    key, actual))
+                        elif val == {'$exists': True}:
+                            if key not in actual:
+                                self.fail("Expected key [%s] in %r" % (
+                                    key, actual))
+                        elif val == {'$exists': False}:
+                            if key in actual:
+                                self.fail("Unexpected key [%s] in %r" % (
+                                    key, actual))
+                        elif key not in actual:
+                            self.fail("Expected key [%s] in %r" % (
+                                key, actual))
+                        else:
+                            self.assertEqual(val, actual[key],
+                                             "Key [%s] in %s" % (key, actual))
+                else:
+                    self.assertEqual(actual, expected)
+
+
+def create_test(scenario_def, test):
+    def run_scenario(self):
+        listener = OvertCommandListener()
+        # New client, to avoid interference from pooled sessions.
+        # Convert test['clientOptions'] to dict to avoid a Jython bug using "**"
+        # with ScenarioDict.
+        client = rs_client(event_listeners=[listener],
+                           **dict(test.get('clientOptions', {})))
+        # Close the client explicitly to avoid having too many threads open.
+        self.addCleanup(client.close)
+
+        # Get database and collection objects.
+        database = getattr(
+            client, scenario_def.get('database_name', TEST_DB))
+        drop_collections(database)
+        collection = getattr(
+            database, scenario_def.get('collection_name', TEST_COLLECTION))
+
+        # Populate collection with data and run test.
+        collection.insert_many(scenario_def.get('data', []))
+        listener.results.clear()
+        self.run_operation(collection, test)
+
+        # Assert expected events.
+        self.check_events(test.get('expectations', {}), listener)
+
+        # Assert final state is expected.
+        expected_outcome = test.get('outcome', {}).get('collection')
+        if expected_outcome is not None:
+            collname = expected_outcome.get('name')
+            if collname is not None:
+                o_collection = getattr(database, collname)
+            else:
+                o_collection = collection
+            self.assertEqual(list(o_collection.find()),
+                             expected_outcome['data'])
+
+    return run_scenario
+
+
+def create_tests():
```

Comment:
It would be great if you could move this `create_tests` into `test/utils.py` and reuse it in both crud spec runners. For now it can simply accept two arguments: the `create_test` method and the `TestAllScenarios` class.